core/src/main/java/org/hisp/dhis/android/core/sms/data/LocalDbRepositoryImpl.java

```diff
@@ -130,4 +133,41 @@ public Completable setMetadataIds(final Metadata metadata) {
                         context.openFileOutput(METADATA_FILE, Context.MODE_PRIVATE), metadata
                 ));
     }
+
+    @Override
+    public Single<Event> getEventToSubmit(String eventUid, String teiUid) {
+        return Single.fromCallable(() ->
```

Comment:
I think this can all be replaced by:

```
Single.fromCallable(() -> eventModule.events.uid(eventUid).withTrackedEntityDataValues())
```

The teiUid should also be set.